cmake-modules/FolderList.cmake

```diff
@@ -72,15 +72,32 @@ macro(DownloadDepVersion DEP_FOLDER DEP_NAME DEP_VERSION)
     set(DOWNLOAD_FILE ${DEP_NAME}_${DEP_VERSION}.zip)
     set(DEP_PREFIX azure-sdk-for-cpp)
 
-    if(FETCH_SOURCE_DEPS STREQUAL "LATEST")
-        message("Downloading latest version of ${DEP_NAME}")
-        #get the latest version from main
-        file(DOWNLOAD http://github.com/Azure/azure-sdk-for-cpp/archive/main.zip ${DOWNLOAD_FOLDER}/${DOWNLOAD_FILE})
-    else()
-        message("Downloading version ${DEP_VERSION} of ${DEP_NAME}")
-        # get the zip
-        file(DOWNLOAD https://github.com/Azure/azure-sdk-for-cpp/archive/refs/tags/${DOWNLOAD_FILE} ${DOWNLOAD_FOLDER}/${DOWNLOAD_FILE})
-    endif()
+    foreach(RETRY_ATTEMPT RANGE 2)
+        if(FETCH_SOURCE_DEPS STREQUAL "LATEST")
+            message("Downloading latest version of ${DEP_NAME}")
+            #get the latest version from main
+            file(
+                DOWNLOAD http://github.com/Azure/azure-sdk-for-cpp/archive/main.zip
+                ${DOWNLOAD_FOLDER}/${DOWNLOAD_FILE}
+                SHOW_PROGRESS
+                STATUS DOWNLOAD_STATUS
+            )
+        else()
+            message("Downloading version ${DEP_VERSION} of ${DEP_NAME}")
+            # get the zip
+            file(
+                DOWNLOAD https://github.com/Azure/azure-sdk-for-cpp/archive/refs/tags/${DOWNLOAD_FILE}
+                ${DOWNLOAD_FOLDER}/${DOWNLOAD_FILE}
+                SHOW_PROGRESS
+                STATUS DOWNLOAD_STATUS
+            )
+        endif()
+
+        list(GET DOWNLOAD_STATUS 0 STATUS_CODE)
+        if (${STATUS_CODE} EQUAL 0)
+            break()
+        endif()
```

Comment:
This could log `DOWNLOAD_STATUS` when there is a nonzero `STATUS_CODE` and if all retries fail throw a fatal error that halts the build. 

Also a delay is advisable in retry scenarios. There appears to be a wonky cmake way to achieve this... https://cmake.org/cmake/help/latest/manual/cmake.1.html#cmdoption-cmake-E-arg-sleep

Reply:
Thanks! The new commit has it all. It does actually wait for the said amount of seconds.
```
1> [CMake] Downloading latest version of 
1> [CMake] -- [download 0% complete]
1> [CMake] Download failed with status code 22: "HTTP response code said error".
1> [CMake] Waiting for 10 seconds before retrying download.
1> [CMake] Downloading latest version of 
1> [CMake] -- [download 0% complete]
1> [CMake] Download failed with status code 22: "HTTP response code said error".
1> [CMake] Waiting for 20 seconds before retrying download.
1> [CMake] Downloading latest version of 
1> [CMake] -- [download 0% complete]
1> [CMake] Download failed with status code 22: "HTTP response code said error".
1> [CMake] CMake Error at C:\src\azure-sdk-for-cpp\CMakeLists.txt:58 (message):
1> [CMake]   Dependency download failed (Link:
1> [CMake]   http://github.com/Azure/azure-sdk-for-cpp/archive/main1.zip).
```